MSBuild reports its version being '0.0.0.0'

## Steps to reproduce
* Get new dotnet/cli preview3 build
* Any msbuild-based command (build)

## Expected behaviour
When MSBuild display it's version, i expect to see a non-zero version number

## Actual behaviour
```
Martins-MacBook-Pro:~ martin$ Documents/dotnet/cli/artifacts/osx.10.11-x64/stage2/dotnet msbuild -help
Microsoft (R) Build Engine version 0.0.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Syntax:              MSBuild.exe [options] [project file]
```
```
Martins-MacBook-Pro:~ martin$ Documents/dotnet/cli/artifacts/osx.10.11-x64/stage2/dotnet build3
Microsoft (R) Build Engine version 0.0.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
```
## Environment
macOS 10.12, pinned RID to `osx.10.11-x64` via `DOTNET_RUNTIME_ID` env var
```
Martins-MacBook-Pro:~ martin$ Documents/dotnet/cli/artifacts/osx.10.11-x64/stage2/dotnet --info
.NET Command Line Tools (1.0.0-preview3-003913)

Product Information:
 Version:            1.0.0-preview3-003913
 Commit SHA-1 hash:  967957698e

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.11-x64
```

Hmm, seems to have regressed

```shell-session
S:\work\GetTargetFrameworkProperties>dotnet\dotnet.exe build3
Microsoft (R) Build Engine version 15.1.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.

S:\work\GetTargetFrameworkProperties>dotnet --version
1.0.0-preview2-003133
```

@jeffkl Think this might be due to the new packages? Maybe versioning isn't right for .NET Core builds produced by installed MSBuild 14?

--- a/src/XMakeBuildEngine/Definition/ProjectCollection.cs
+++ b/src/XMakeBuildEngine/Definition/ProjectCollection.cs
@@ -403,7 +403,7 @@ namespace Microsoft.Build.Evaluation
                     // work when Microsoft.Build.dll has been shadow-copied, for example
                     // in scenarios where NUnit is loading Microsoft.Build.
                     var versionInfo = FileVersionInfo.GetVersionInfo(FileUtilities.ExecutingAssemblyPath);
-                    s_engineVersion = new Version(versionInfo.FileMajorPart, versionInfo.FileMinorPart, versionInfo.FileBuildPart, versionInfo.FilePrivatePart);
+                    s_engineVersion = new Version(versionInfo.ProductMajorPart, versionInfo.ProductMinorPart, versionInfo.ProductBuildPart, versionInfo.ProductPrivatePart);
                 }
 
                 return s_engineVersion;

--- a/src/XMakeBuildEngine/UnitTestsPublicOM/Definition/ProjectCollection_Tests.cs
+++ b/src/XMakeBuildEngine/UnitTestsPublicOM/Definition/ProjectCollection_Tests.cs
@@ -1427,6 +1427,21 @@ namespace Microsoft.Build.UnitTests.OM.Definition
         }
 
         /// <summary>
+        /// Verifies that the <see cref="ProjectCollection.Version"/> is correct.
+        /// </summary>
+        /// <remarks>
+        /// This test was written because on OS X and Ubuntu, the version is showing "0.0.0.0"
+        /// while Windows was working just fine.
+        /// </remarks>
+        [Fact]
+        public void ProjectCollectionVersionIsCorrect()
+        {
+            Version expectedVersion = new Version(MSBuildConstants.CurrentAssemblyVersion);
+
+            Assert.Equal(expectedVersion, ProjectCollection.Version);
+        }
+
+        /// <summary>
         /// Create an empty project file and return the path
         /// </summary>
         private static string CreateProjectFile()

